dev-tools-java/src/main/java/org/jboss/forge/addon/devtools/java/JavaFormatSourcesCommand.java

```diff
@@ -0,0 +1,169 @@
+import org.jboss.forge.addon.ui.context.UIExecutionContext;
+import org.jboss.forge.addon.ui.input.InputComponentFactory;
+import org.jboss.forge.addon.ui.input.UIInput;
+import org.jboss.forge.addon.ui.input.UIInputMany;
+import org.jboss.forge.addon.ui.metadata.UICommandMetadata;
+import org.jboss.forge.addon.ui.result.Result;
+import org.jboss.forge.addon.ui.result.Results;
+import org.jboss.forge.addon.ui.util.Categories;
+import org.jboss.forge.addon.ui.util.Metadata;
+import org.jboss.forge.furnace.Furnace;
+import org.jboss.forge.furnace.container.simple.lifecycle.SimpleContainer;
+import org.jboss.forge.roaster.Roaster;
+
+public class JavaFormatSourcesCommand extends AbstractUICommand
+{
+
+   private final InputComponentFactory inputFactory;
+
+   private UIInput<XMLResource> profile;
+
+   @SuppressWarnings("rawtypes")
+   private UIInputMany<FileResource> sources;
+
+   public JavaFormatSourcesCommand()
+   {
+
+      Furnace furnace = SimpleContainer.getFurnace(this.getClass().getClassLoader());
+      this.inputFactory = furnace.getAddonRegistry().getServices(InputComponentFactory.class).get();
+
+   }
+
+   @Override
+   public UICommandMetadata getMetadata(UIContext context)
+   {
+      return Metadata
+               .forCommand(getClass())
+               .name("Java: Format Sources")
+               .description(
+                        "command to format a file (or folder recursively) of Java source files")
+               .category(Categories.create("Java"));
+
+   }
+
+   @Override
+   public void initializeUI(UIBuilder builder) throws Exception
+   {
+
+      sources = inputFactory.createInputMany("sources", 's', FileResource.class);
+      sources.setRequired(true);
+      sources.setDescription("The folder or file where the java sources will be formatted");
+
+      profile = inputFactory.createInput("profile", 'p', XMLResource.class);
+      profile.setDescription("The eclipse code format file");
+
+      builder.add(sources).add(profile);
+
+   }
+
+   @SuppressWarnings("rawtypes")
+   @Override
+   public Result execute(UIExecutionContext context) throws Exception
+   {
+
+      XMLResource formatProfileLocation = profile.getValue();
+      Iterable<FileResource> formatSources = sources.getValue();
+
+      List<FileResource<?>> fileResourceList = new ArrayList<>();
+
+      for (FileResource<?> fileResource : formatSources)
+         fileResourceList.add(fileResource);
+
+      if (formatProfileLocation == null)
+         format(fileResourceList);
+
+      else
+      {
+         Properties formatProfile = new Properties();
+         formatProfile.load(formatProfileLocation.getResourceInputStream());
+         format(fileResourceList, formatProfile);
+      }
+
+      return Results.success("Files Formatted Sucessfully");
+
+   }
+
+   // Formatting the file or folder(recursively) with custom profile input.
+   private static void format(List<FileResource<?>> fileResourceList, Properties formatProfile)
```

Comment:
Better merge both format methods into one only. Just pass null to the formatProfile in the case where the profile XML is not provided